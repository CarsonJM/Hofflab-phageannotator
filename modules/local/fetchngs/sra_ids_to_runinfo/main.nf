
process SRA_IDS_TO_RUNINFO {
    tag "${meta.id}"
    label 'error_retry'

    conda "conda-forge::python=3.9.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    tuple val(meta), val(id)
    val fields

    output:
    tuple val(meta), path("*.tsv")  , emit: tsv
    path "versions.yml"             , emit: versions

    script:
    def metadata_fields = fields ? "--ena_metadata_fields ${fields}" : ''
    """
    echo $id > id.txt
    sra_ids_to_runinfo.py \\
        id.txt \\
        ${id}.runinfo.tsv \\
        $metadata_fields

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """
}
