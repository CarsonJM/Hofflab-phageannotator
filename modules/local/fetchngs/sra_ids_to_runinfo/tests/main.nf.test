nextflow_process {

    name "Test process: SRA_IDS_TO_RUNINFO"
    script "../main.nf"
    process "SRA_IDS_TO_RUNINFO"

    test("SRR13191702") {

        when {
            process {
                """
                input[0] = 'SRR13191702'
                input[1] = ''
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("SRR13191702 - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = 'SRR13191702'
                input[1] = ''
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
