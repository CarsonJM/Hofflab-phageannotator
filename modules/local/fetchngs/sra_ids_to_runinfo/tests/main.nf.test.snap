{
    "SRR13191702": {
        "content": [
            {
                "0": [
                    "SRR13191702.runinfo.tsv:md5,86ca7bdf5452889f99961e03e4ddf5b8"
                ],
                "1": [
                    "versions.yml:md5,1c14442e9b494b586eafe41e77300fae"
                ],
                "tsv": [
                    "SRR13191702.runinfo.tsv:md5,86ca7bdf5452889f99961e03e4ddf5b8"
                ],
                "versions": [
                    "versions.yml:md5,1c14442e9b494b586eafe41e77300fae"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-22T10:42:16.307943246"
    },
    "SRR13191702 - stub": {
        "content": [
            {
                "0": [
                    "SRR13191702.runinfo.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "versions.yml:md5,1c14442e9b494b586eafe41e77300fae"
                ],
                "tsv": [
                    "SRR13191702.runinfo.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,1c14442e9b494b586eafe41e77300fae"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-22T10:42:22.864623928"
    }
}